ann_hw/src/neuralnet.py

```diff
@@ -0,0 +1,119 @@
+#! /usr/bin/env python
+
+from __future__ import print_function
+from __future__ import division
+
+import sys
+import arff
+import numpy as np
+import pandas as pd
+from tools import *
+import csv
+
+def train_nn(num_epochs, num_folds, learning_rate, w_ij, w_jk, data):
+    """
+    Given the parameters obtained by user CLI input, initialized random weights, 
+    and the training data, this loops through the entire data set to calculate
+    the weights given to each node in each layer. This algorithm is written for 
+    a single hidden layer of equal size to the input layer, a bias node on each, 
+    and assumes binary classification.
+    """
+    for epoch in range(0, num_epochs):
+        strat_sets, classes = strat_split(num_folds, data)
+        tests = {}
+        for f in range(1, num_folds+1):
+            # set aside a testing set used during prediciton phase
+            test = strat_sets.pop(f)
+            tests[f] = test
+            for fold, fold_set in strat_sets.iteritems():
+                for instance, row in fold_set.iterrows():
+                ####################
+                ### feed forward ###
```

Comment:
I think if all these variables were in a class then I could make separate feed forward and backpropagate fns?